homeassistant/components/sensor/meteo_france.py

```diff
@@ -0,0 +1,148 @@
+    meteofrance_data = MeteoFranceCurrentData(hass, location_id)
+
+    try:
+        meteofrance_data.update()
+    except ValueError as err:
+        _LOGGER.error("Received error from Meteo France: %s", err)
+        return False
+
+    add_devices([MeteoFranceSensor(meteofrance_data, location_name)])
+
+
+class MeteoFranceSensor(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, meteofrance_data, location_name):
+        """Initialize the sensor."""
+        self.current_data = meteofrance_data
+        self.location_name = location_name
+        self._unit = "Min"
+        self._time_to_rain = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        if self.location_name is None:
+            return "1hr rain forecast"
+        return "{} 1hr rain forecast".format(self.location_name)
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        if self.current_data and self.current_data.rain_forecast:
+            self._time_to_rain = 0
+            for interval in self.current_data.intervals:
+                if interval["niveauPluie"] > 1:
+                    self._unit = "Min"
+                    return self._time_to_rain
+                self._time_to_rain += 5
+            self._unit = ""
+            return "No rain"
+        return STATE_UNKNOWN
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes of the sun."""
+        if self.current_data and self.current_data.rain_forecast:
+            return {
+                **{
+                    STATE_ATTR_FORECAST: self.current_data.rain_forecast,
+                },
+                **{
+                    STATE_ATTR_FORECAST_INTERVAL+str(interval+1):
+                        self.current_data.intervals[interval]["niveauPluie"]
+                    for interval in range(0, len(self.current_data.intervals))
+                },
+                **{
+                    ATTR_ATTRIBUTION: CONF_ATTRIBUTION
+                }
+            }
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._unit
+
+    def update(self):
+        """Fetch new state data for the sensor."""
+        self.current_data.update()
+
+
+class MeteoFranceCurrentData():
+    """Get data from Meteo France."""
+
+    def __init__(self, hass, location_id):
+        """Initialize the data object."""
+        self._hass = hass
+        self._location_id = location_id
+        self.rain_forecast = None
+        self.intervals = None
+
+    def _build_url(self):
+        url = _RESOURCE.format(self._location_id)
+        _LOGGER.info("Meteo France rain forecast URL %s", url)
+        return url
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """Get the latest data from BOM."""
+        try:
+            result = requests.get(self._build_url(), timeout=10).json()
```

Comment:
Please extract service specific I/O and data parsing to a standalone library published on pypi.

https://developers.home-assistant.io/docs/en/creating_platform_code_review.html#6-communication-with-devices-services